### PR TITLE
[16.0] [MIG] base_export_manager: Migration to 16.0

### PR DESCRIPTION
**Pull request by rjpathan**

### CODE REVIEW

**Comment by kikopeiro** [comment]
Here's what I tried:

- I changed the configuration (Accesses Rights) for the res.partner.category  (all accesses)
- I still can access the export option over this model
- Tested with not admin user

Did I miss something? V15 is working fine

What is reproduced is similar to what was discussed in this issue https://github.com/OCA/server-ux/issues/474.

**Comment by rjpathan** [comment]
@kikopeiro sorry for late reply.

Export action button is disabled with a javascript function https://github.com/rjpathan/server-ux/blob/16.0-mig-base_export_manager/base_export_manager/static/src/js/base_export_manager.js
It seems this is not working in v16.

Sorry, I am not sure how to fix this as I don't have proficiency with the javascript.

**Comment by hugosantosred** [comment]
@rjpathan I've opened a [PR](https://github.com/rjpathan/server-ux/pull/1) in your branch with the fix. In v16 the ListController is defined in OWL, so is needed to patch that controller in order to set the isExportEnable property

**Comment by rjpathan** [comment]
@hugosantosred Thanks for the fix.

File: base_export_manager/models/ir_exports_line.py [hunk 1]
```diff
@@ -142,8 +142,8 @@ def _inverse_name(self):
             if any(parts):
                 # invalidate_cache -> in order to get actual value of field 'label'
                 # in function '_check_name'
-                self.invalidate_cache(ids=one.ids)
-                one._check_name()
+                self.env.flush_all()
+                self.env.invalidate_all()
```
Comment (stefan-tecnativa) [inline base_export_manager/models/ir_exports_line.py:146]: Passing `flush=True` parameter to invalidate_all() wouldn't be the same as both lines that you wrote?
https://github.com/odoo/odoo/blob/16.0/odoo/api.py#L714-L723

**Comment by stefan-tecnativa** [comment]
You're pre-commit problem is related with the list_controller.js file. Two possible solutions could be:
1- Convert your code to ESM module (ECMAScript Modules)
2- Use requires to import objects from other files: https://github.com/OCA/OCB/blob/15.0/addons/account/static/src/js/account_payment_field.js#L5

Good Luck!

**Review by kikopeiro — APPROVED** [review]
:+1:  LGTM (without code review)

File: base_export_manager/models/ir_exports_line.py [hunk 2]
```diff
@@ -142,8 +142,7 @@ def _inverse_name(self):
             if any(parts):
                 # invalidate_cache -> in order to get actual value of field 'label'
                 # in function '_check_name'
-                self.invalidate_cache(ids=one.ids)
-                one._check_name()
+                self.env.invalidate_all()
```
Comment (stefan-tecnativa) [inline base_export_manager/models/ir_exports_line.py:145]: self.browse(one.ids).invalidate_recordset()
one._check_name()
```

**Review by stefan-tecnativa — CHANGES_REQUESTED** [review]

**Comment by pedrobaeza** [comment]
Superseded by #702

**Pull request closed without merging**